.github/workflows/build-and-push-image.yml

```diff
@@ -1,12 +1,16 @@
 name: Deploy to environment (CIP)
 
 on:
+  push:
+    branches:
+      - main
   workflow_dispatch:
     inputs:
       environment:
         type: environment
         description: "Choose an environment to deploy to"
         required: true
+        default: "Dev"
```

Comment:
This should mean the workflow runs on every push to `main` which will automate your deployment to Dev but also allows operators to select using workflow_dispatch which target environment to deploy to (Staging/ Prod)